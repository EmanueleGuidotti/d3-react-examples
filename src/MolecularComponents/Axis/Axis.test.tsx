import React from "react";
import { Axis } from "./index";
import { render } from "@testing-library/react";
import * as d3 from "d3";

describe("Axis", () => {
  it("create Axis Left with values from 0 to 10", () => {
    const container = render(
      <svg width="800" height="400" id="svg">
        <Axis
          scaleFunction={d3
            .scaleLinear()
            .domain([0, 10])
            .range([0, 200])}
          axisFunction={d3.axisLeft}
        />
      </svg>,
    );

    expect(
      document.getElementsByClassName("tick").length,
    ).toBe(11);

    expect(
      document.getElementsByTagName("text").length,
    ).toBe(11);

    const texts = Array.from(
      document.getElementsByTagName("text"),
    );

    texts.forEach((text, index) => {
      expect(text.innerHTML).toBe(String(index));
    });

    expect(container).toMatchSnapshot();
  });
  it("create Axis Bottom with values from 0 to 10", () => {
    const container = render(
      <svg width="800" height="400" id="svg">
        <Axis
          scaleFunction={d3
            .scaleLinear()
            .domain([0, 10])
            .range([0, 200])}
          axisFunction={d3.axisBottom}
        />
      </svg>,
    );

    expect(
      document.getElementsByClassName("tick").length,
    ).toBe(11);

    expect(
      document.getElementsByTagName("text").length,
    ).toBe(11);

    const texts = Array.from(
      document.getElementsByTagName("text"),
    );

    texts.forEach((text, index) => {
      expect(text.innerHTML).toBe(String(index));
    });

    expect(container).toMatchSnapshot();
  });
});
