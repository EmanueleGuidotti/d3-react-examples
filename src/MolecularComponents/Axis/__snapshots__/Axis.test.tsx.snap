// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Axis create Axis Bottom with values from 0 to 10 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        height="400"
        id="svg"
        width="800"
      >
        <g
          fill="none"
          font-family="sans-serif"
          font-size="10"
          text-anchor="middle"
          transform="translate(10, 10)"
        >
          <path
            class="domain"
            d="M0.5,6V0.5H500.5V6"
            stroke="currentColor"
          />
          <g
            class="tick"
            opacity="1"
            transform="translate(0.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(50.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              1
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(100.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              2
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(150.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              3
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(200.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              4
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(250.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(300.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              6
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(350.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              7
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(400.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              8
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(450.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              9
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(500.5,0)"
          >
            <line
              stroke="currentColor"
              y2="6"
            />
            <text
              dy="0.71em"
              fill="currentColor"
              y="9"
            >
              10
            </text>
          </g>
        </g>
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      height="400"
      id="svg"
      width="800"
    >
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(10, 10)"
      >
        <path
          class="domain"
          d="M0.5,6V0.5H500.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(50.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            1
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(100.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            2
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(150.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            3
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(200.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            4
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(250.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(300.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            6
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(350.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            7
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(400.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            8
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(450.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            9
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(500.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            10
          </text>
        </g>
      </g>
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Axis create Axis Left with values from 0 to 10 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        height="400"
        id="svg"
        width="800"
      >
        <g
          fill="none"
          font-family="sans-serif"
          font-size="10"
          text-anchor="end"
          transform="translate(10, 10)"
        >
          <path
            class="domain"
            d="M-6,0.5H0.5V200.5H-6"
            stroke="currentColor"
          />
          <g
            class="tick"
            opacity="1"
            transform="translate(0,0.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              0
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,20.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              1
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,40.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              2
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,60.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              3
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,80.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              4
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,100.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              5
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,120.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              6
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,140.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              7
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,160.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              8
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,180.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              9
            </text>
          </g>
          <g
            class="tick"
            opacity="1"
            transform="translate(0,200.5)"
          >
            <line
              stroke="currentColor"
              x2="-6"
            />
            <text
              dy="0.32em"
              fill="currentColor"
              x="-9"
            >
              10
            </text>
          </g>
        </g>
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      height="400"
      id="svg"
      width="800"
    >
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(10, 10)"
      >
        <path
          class="domain"
          d="M-6,0.5H0.5V200.5H-6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,20.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            1
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,40.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            2
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,60.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            3
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,80.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            4
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,100.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,120.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            6
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,140.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            7
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,160.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            8
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,180.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            9
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,200.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            10
          </text>
        </g>
      </g>
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
